ams/server/src/main/java/com/netease/arctic/server/optimizing/TaskRuntime.java

```diff
@@ -299,7 +300,7 @@ public String toString() {
 
   private void validThread(OptimizerThread thread) {
     if (token == null) {
-      throw new IllegalStateException("Task not scheduled yet, taskId:" + taskId);
+      throw new IllegalTaskException("Task has been reset or not yet scheduled, taskId:" + taskId);
```

Comment:
Good job, the error description is more proper now.

But we may improve the error message in line 305 too, like:
```
The optimizer thread does not match, (xxx) in the task, (xxx) in the request.
```